x/mock/app.go

```diff
@@ -43,7 +46,7 @@ func NewApp() *App {
 
 	// Create your application object
 	app := &App{
-		BaseApp:    bam.NewBaseApp("mock", cdc, logger, db),
+		BaseApp:    bam.NewBaseApp("mock", cdc, ctx, db),
```

Comment:
This doesn't make a whole lot of sense to me, I think of context as something created by bass to facilitate information pertaining to the block... unintuitive to create this as a part of `NewApp` it's not dealing with blocks.